src/node_external_reference.h

```diff
@@ -22,6 +22,9 @@ using CFunctionCallbackWithBool = void (*)(v8::Local<v8::Object> receiver,
 using CFunctionCallbackWithStrings =
     bool (*)(v8::Local<v8::Value>, const v8::FastOneByteString& input);
 
+using CFunctionCallbackWithStringsReturnVoid =
+    void (*)(v8::Local<v8::Value>, const v8::FastOneByteString& input);
```

Comment:
@joyeecheung You used `Local<v8::Object>` for the receiver, whereas, in `CanParse` @KhafraDev used `Local<v8::Value>` for the receiver. Which one is the correct usage for the Fast API, or is both of them correct?

Reply:
The strings returned by `createObjectURL()` [are cons strings](https://github.com/nodejs/node/blob/50d510f0de546c9fda42dbce6114b2b35449d399/lib/internal/url.js#L1045). They won't hit the fast API.

Reply:
Is this documented? How can we update the benchmark to flatten the string?

(For people who didn't know what `cons strings` mean: cons strings are pairs of strings, result of concatenation.)

Reply:
Adding `hello-1-2-3` as an input triggers fast api, but even `crypto.randomUUID()` as a parameter does not trigger fast API...

```
function main({ n }) {
  bench.start();
  for (let i = 0; i < n; i += 1) {
    URL.revokeObjectURL('hello-1-2-3');
  }
  bench.end(n);
}
```

Reply:
`FastOneByteString` means the string type is a sequential one byte string. The string types are internal to V8 and subject to changes. I don't think it's meaningful to flatten the string in the benchmark if users in the wild are going to pass urls returned by `createObjectURL()` to it...

Reply:
Would it make sense to flatten createObjectURL response?

Reply:
What triggers flattening is also internal to V8 heuristics. Although in the case of `createObjectURL()` I wonder why we can't just generate the UUID from C++ and construct the string from C++ too...

Reply:
hmmm i thought v8 would just insert a flatten node for fast call string arguments, did i hallucinate that?

Reply:
Actually if we generate the UUID from C++ we can probably implement the heavy-lifting of `createObjectURL()` in a fast call, because the arguments are simple enough and we don't actually need heap allocations in it.

Reply:
Oh wait, the returned value needs heap allocation. Oh well. But I think we can still just implement it in a normal C++ binding, which can probably just be merged into `Blob::StoreDataObject` because that's the only place it's used.

Reply:
I there a C++ function implemented for generating UUIDs?

Reply:
I am giving try to just construct the final string on C++ side to see if it triggers fast calls

Reply:
> I there a C++ function implemented for generating UUIDs?

I think this could be used - https://github.com/nodejs/node/blob/b5fe45fb9a2fb30432bc446ec87cc2780f2ff679/src/crypto/crypto_util.h#L126

Reply:
Ok i tried building the string in C++ in 268fc13 , the fast calls are triggered i confirmed but turns out its not fast heh its ~2% slower it seems, maybe because adding the string allocation slows down `createObjectURL`

Reply:
Also I tried using CSPRNG function to make the UUID in C++ side, found that makes things even more slower, I wrote a function like this one here https://github.com/nodejs/node/blob/268fc131b9ef99fcd60c9aa6668834d3d41d450a/src/inspector_io.cc#L45-L63

Reply:
The JS-land version buffers the UUID so to match the performance the C++ version needs to buffer the UUID as well. Also `snprintf()` can be slow if it's heavily used.

Reply:
I think doing all this would probably make the C++ side way more complicated than the original TODO intended, is there no faster way of string allocation on C++ side?

Reply:
You could also just merge the concatenation in `createObjectURL()` from JS land into `Blob::StoreDataObject` and concatenate the strings from C++ instead.

Reply:
The string should be created with `NewFromOneByte()`, not `NewFromUtf8()` as in https://github.com/nodejs/node/commit/268fc131b9ef99fcd60c9aa6668834d3d41d450a because it's guaranteed to be one-byte and does not need transcoding.

Reply:
okay so i tried this 

```diff
diff --git a/src/node_blob.cc b/src/node_blob.cc
index 7db8684904..45242e7c44 100644
--- a/src/node_blob.cc
+++ b/src/node_blob.cc
@@ -391,13 +391,19 @@ void Blob::StoreDataObject(const v8::FunctionCallbackInfo<v8::Value>& args) {
 
   size_t length = args[2].As<Uint32>()->Value();
   Utf8Value type(env->isolate(), args[3]);
+  std::string key_str(*key, key.length());
 
   binding_data->store_data_object(
-      std::string(*key, key.length()),
-      BlobBindingData::StoredDataObject(
-        BaseObjectPtr<Blob>(blob),
-        length,
-        std::string(*type, type.length())));
+      key_str,
+      BlobBindingData::StoredDataObject(BaseObjectPtr<Blob>(blob),
+                                        length,
+                                        std::string(*type, type.length())));
+  std::string final_url = "blob:nodedata:" + key_str;
+  args.GetReturnValue().Set(String::NewFromOneByte(env->isolate(),
+                                                reinterpret_cast<const uint8_t*>(final_url.data()),
+                                                v8::NewStringType::kNormal,
+                                                final_url.length())
+                                .ToLocalChecked());
 }
```

fast paths are being hit and perf is maybe little bit better than https://github.com/nodejs/node/commit/268fc131b9ef99fcd60c9aa6668834d3d41d450a but on the benchmark in the commit, its still slower than the normal js version unfortunately 😓